### PR TITLE
Close the OkHttp response in case of error. Fixes https://github.com/…

### PR DESCRIPTION
**Pull request by kypeli**
## Description
Closes OkHttp and OkHttp3 Responses in case of error. Fixes issue https://github.com/bumptech/glide/issues/1441

### CODE REVIEW

**Comment by kypeli** [comment]
Happened to see this trivial issue in Glide, so I hope my PR is useful. Didn't notice any tests related to OkHttp in Glide, but I am very new to the Glide source code, so please correct me if I am wrong. 

Anything else that is needed for this?

File: integration/okhttp3/src/main/java/com/bumptech/glide/integration/okhttp3/OkHttpStreamFetcher.java [hunk 1]
```diff
@@ -58,6 +58,7 @@ public void onResponse(Call call, Response response) throws IOException {
           stream = ContentLengthInputStream.obtain(responseBody.byteStream(), contentLength);
         } else if (Log.isLoggable(TAG, Log.DEBUG)) {
           Log.d(TAG, "OkHttp got error response: " + response.code() + ", " + response.message());
+          response.body().close();
```
Comment (TWiStErRob) [inline integration/okhttp3/src/main/java/com/bumptech/glide/integration/okhttp3/OkHttpStreamFetcher.java]: What if logging is disabled?

**Comment by TWiStErRob** [comment]
Thanks, there's only one issue: it's a partial fix, see comment on code.

I think to keep it compatible (similar) with v3 it would be better to assign `responseBody` as soon as there is one. `cleanup` is called in a ` finally` as far as I know.

**Comment by kypeli** [comment]
You are absolutely right, that's a lot better. Thanks. I updated the pull request as suggested.

**Comment by TWiStErRob** [comment]
Tip for future: you don't need the full url in the commit message `#1234` or `user/repo#1234` works as well.

**Comment by sjudd** [comment]
Thanks!

**Pull request merged**